homeassistant/components/myuplink/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""The myUplink integration."""
+from __future__ import annotations
+
+from myuplink.api import MyUplinkAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import (
+    aiohttp_client,
+    config_entry_oauth2_flow,
+    device_registry as dr,
+)
+
+from . import api
+from .const import (
+    DOMAIN,
+    MU_DATAGROUP_DEVICES,
+    MU_DEVICE_FIRMWARE_CURRENT,
+    MU_DEVICE_PRODUCTNAME,
+)
+from .coordinator import MyUplinkDataCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up myUplink from a config entry."""
+
+    hass.data[DOMAIN] = {}
```

Comment:
I could be missing something/ be unaware of something though?

Reply:
It's "normal" what i can see, and i got some problems (i could be wrong) if i removed that line.

Reply:
We normally use `dict.setdefault`. See other integrations for examples. We shouldn't reset the hass.data since there could potentially be more than one config entry for this integration.